system/metrics/sinks/file/fileSink.hpp

```diff
@@ -31,27 +27,21 @@ using namespace hpccMetrics;
 #define FILESINK_API DECL_IMPORT
 #endif
 
-class FILESINK_API FileMetricSink : public MetricSink
+class FILESINK_API FileMetricSink : public PeriodicMetricSink
 {
-    public:
-        explicit FileMetricSink(const char *name, const IPropertyTree *pSettingsTree);
-        ~FileMetricSink() override;
-        void startCollection(MetricsReporter *pReporter) override;
-        void stopCollection() override;
-
-    protected:
-        void writeReportHeaderToFile() const;
-        void writeMeasurementToFile(const std::string &metricName, uint32_t value, const std::string &metricDescription) const;
-        void collectionThread();
-        void doStopCollecting();
-
-    protected:
-        StringBuffer fileName;
-        unsigned collectionPeriodSeconds;
-        std::thread collectThread;
-        std::atomic<bool> stopCollectionFlag{false};
-        bool isCollecting = false;
-        bool clearFileOnStartCollecting = false;
-        FILE *fhandle = nullptr;
-        Semaphore waitSem;
+public:
+    explicit FileMetricSink(const char *name, const IPropertyTree *pSettingsTree);
+    ~FileMetricSink() override = default;
+
+protected:
+    void prepareToStartCollecting() override;
```

Comment:
minor: As a general style, we always include the virtual keyword - even when it isn't needed.  Primarily to improve comprehension.